cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -570,6 +570,16 @@ bool IsContainerNotFound(const Storage::StorageException& e) {
   return false;
 }
 
+bool IsForbidden(const Storage::StorageException& e) {
+  // In some situations, only the ReasonPhrase is set and the
+  // ErrorCode is empty, so we check both.
+  if (e.ErrorCode == "AuthorizationFailure" ||
+      e.ReasonPhrase == "This request is not authorized to perform this operation.") {
+    DCHECK_EQ(e.StatusCode, Http::HttpStatusCode::Forbidden);
+  }
```

Comment:
Can we remove this?

```suggestion
```

Reply:
I don't know. I copied that code from the function directly above, copied below. Happy to remove it, but I figured someone wrote it that way for a reason.

bool IsContainerNotFound(const Storage::StorageException& e) {
  // In some situations, only the ReasonPhrase is set and the
  // ErrorCode is empty, so we check both.
  if (e.ErrorCode == "ContainerNotFound" ||
      e.ReasonPhrase == "The specified container does not exist." ||
      e.ReasonPhrase == "The specified filesystem does not exist.") {
    DCHECK_EQ(e.StatusCode, Http::HttpStatusCode::NotFound);
    return true;
  }
  return false;
}

Reply:
In the case, 404 Not Found may be returned not for cases that container/filesystem doesn't exist too. So we can't use `Http::HttpStatusCode::NotFound` directly.